icicle/primitives/field.cuh

```diff
@@ -38,24 +38,115 @@ public:
+      {0xc78c8967, 0x6fdd00bf, 0x434906ac, 0x146b58bc, 0x972e89ed, 0x2ccddea2, 0x37b1da3d, 0x485d5127},
+      {0x2a5ad9e1, 0x034d2ff2, 0xa9152435, 0xae4622f6, 0x0d477fa6, 0xdc86b01c, 0xb500a166, 0x56624634},
+      {0x1279bb6e, 0xfbd047e1, 0xb3f32699, 0xc8d5f51d, 0x7a0cbe39, 0x48340541, 0xe558b50d, 0x3291357e},
+      {0xcd96b8ad, 0xd7118f85, 0x1fcadc91, 0x67a665ae, 0xf1aeb578, 0x88f39a78, 0x12180caa, 0x2155379d},
+      {0xf3b70f10, 0x08692405, 0xd0711b7d, 0xcd7f2bd6, 0x772c33d6, 0x473a2eef, 0x2d8acbf4, 0x22426233},
+      {0x8ef674fb, 0x6f421a7d, 0x30ce40fd, 0xbb97a3bf, 0xe1c34bb0, 0x652f717a, 0x30794f01, 0x2d3056a5},
+      {0xecb38d9d, 0x194e8c62, 0x4419c750, 0xad8e16e8, 0xd0adef90, 0xdf625e80, 0x724a6955, 0x520e587a},
+      {0x39898d4b, 0xfece7e0e, 0x265e09d9, 0x2f69e02d, 0xcb98de4a, 0xa57a6e07, 0xcb947035, 0x03e1c54b},
+      {0x2d3ea03a, 0xcd397912, 0x04db5844, 0x46b3105f, 0xb0691d4e, 0xc70d0874, 0x7018af4f, 0x47c8b581},
+      {0xb08e3363, 0xc6e7a6ff, 0x86389bee, 0xe08fec7c, 0xfbb8d1bb, 0xf2d38f10, 0x5abcaa32, 0x0abe6a5e},
+      {0xe0ec9eae, 0x5616c57d, 0x585a72db, 0xc631ffb2, 0xa3b51e3c, 0x5121af06, 0xaa0655b2, 0x73560252},
+      {0x77bd779c, 0x92cf4deb, 0x29b7d7bc, 0x72cf6a80, 0xee762730, 0x6e0bcd91, 0x8823e687, 0x291cf6d6},
+      {0x4e11a51e, 0xce32ef84, 0x3da64ca5, 0xc0ba12bb, 0xdc61a1a3, 0x0454dc1e, 0xfd328739, 0x019fe632},
+      {0xd2d75182, 0x531a11a0, 0x02867ddc, 0x02c81184, 0xfbedc11d, 0x116168bf, 0xb1980c0d, 0x0a0a77a3},
+      {0x9f0319ed, 0xe2d0a786, 0xb1d7a628, 0xb94f1101, 0x4f31d25d, 0xece8ea22, 0x00f8f98b, 0x23397a93},
+      {0x0a4f2089, 0xd7b68883, 0xf6ac7b41, 0x6558e9e3, 0x71905a7d, 0x99e276b5, 0x2f094256, 0x52dd465e},
+      {0x9d8e211b, 0x47465035, 0x6214abc6, 0x84d37b82, 0xf2bb4598, 0x8da40c1e, 0x44bf1bee, 0x0c83ea77},
+      {0x08c9dd56, 0x694341f6, 0xabb30adc, 0xed3a181f, 0xda8b398f, 0x1339a815, 0x11657e1e, 0x2c6d4e45},
+      {0x06ffc93f, 0x63e7cb49, 0xe28a193d, 0xf070bb00, 0x2e5713b5, 0xad1715b0, 0x5990693f, 0x4b537149},
+      {0x439f0d2b, 0x3829971f, 0x8c2280b9, 0xb6368350, 0x22c813b4, 0xd09b6819, 0xdfe81f20, 0x16a2a19e}};
+
+    // Quick fix to linking issue, permanent fix will follow
+    switch (log_size) {
+    case 0:
+      return Field{CONFIG::one};
+    default: {
+      return Field{_omega[log_size - 1]};
+    }
+    }
+  }
+
+  // static HOST_INLINE Field omega_inv(uint32_t logn)
+  // {
+  //   if (logn == 0) { return Field{CONFIG::one}; }
+
+  //   if (logn > CONFIG::omegas_count) { throw std::invalid_argument("Field: Invalid omega_inv index"); }
 
-    if (logn > CONFIG::omegas_count) { throw std::invalid_argument("Field: Invalid omega_inv index"); }
+  //   storage_array<CONFIG::omegas_count, TLC> const omega_inv = CONFIG::omega_inv;
+  //   return Field{omega_inv.storages[logn - 1]};
+  // }
 
-    storage_array<CONFIG::omegas_count, TLC> const omega_inv = CONFIG::omega_inv;
-    return Field{omega_inv.storages[logn - 1]};
+  static constexpr HOST_INLINE Field omega_inv(uint32_t log_size)
```

Comment:
hotfix for omegas and omegas to match Arkworks (and other common NTT implementations) - _"correct"_ twiddles are restored to 'pre-ETH' compatibility - like in the old curve generation script, just without [reverse](https://github.com/ingonyama-zk/icicle/blob/fd62fe5ae8504b72c87d125f8a83541a73d1bff3/curve_parameters/new_curve_script.py#L125)

pls note - this will be replaced by customizable twiddle generation like (similar to the script). So the twiddles can be arbitrarily selected and not on compile time. For flexibility and compatibility with wast number of implementations